### PR TITLE
feat: migrated CaDeT data bucket from pulumi into terraform

### PR DESCRIPTION
**Pull request by tom-webber**
[CaDeT data bucket](https://github.com/ministryofjustice/create-a-derived-table-infrastructure/blob/e58b7c512add0f25fc95c390089947294714e107/create_a_derived_table_infra/__main__.py#L126) put into data-engineering-pipelines as terraform

Co-authored-by: Ant Fitzroy <AntFMoJ@users.noreply.github.com>

# Pull Request Objective

This piece of work is being tracked in
[this](https://github.com/ministryofjustice/analytical-platform/issues/5599)
GitHub Issue.

This PR translates the [CaDeT data bucket and bucket policies from pulumi](https://github.com/ministryofjustice/create-a-derived-table-infrastructure/blob/e58b7c512add0f25fc95c390089947294714e107/create_a_derived_table_infra/__main__.py#L126) into terraform.

**Note:** These resources may be better suited within a new `create-a-derived-table` component within `terraform/aws/analytical-platform-data-production` than within `data-engineering-pipelines`.

## Checklist

- [x] I have reviewed the [style guide](https://docs.analytical-platform.service.justice.gov.uk/documentation/platform/infrastructure/terraform.html#terraform)
and ensured that my code complies with it
- [x] All checks have passed (or override label applied, if I've
used the `override-static-analysis` label, I've explained why)
- [x] I have self-reviewed my code
- [x] I have reviewed the checks and can attest they're as expected

### CODE REVIEW

**Review by jhpyke — DISMISSED** [review]
Can you put an [import block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#import) in place for the existing resource so we can see if the plan would lead to divergence from the current state of the bucket? Don't use the CLI import as that'll lock the state, just the inline one.

File: terraform/aws/analytical-platform-data-production/create-a-derived-table/terraform.tfvars [hunk 1]
```diff
@@ -0,0 +1,14 @@
+account_ids = {
+  analytical-platform-data-production       = "593291632749"
+  analytical-platform-management-production = "042130406152"
+}
+tags = {
+  business-unit          = "Platforms"
+  application            = "Data Platform"
```
Comment (jacobwoffenden) [inline terraform/aws/analytical-platform-data-production/create-a-derived-table/terraform.tfvars]:
```suggestion
  application            = "Analytical Platform"
```

File: terraform/aws/analytical-platform-data-production/create-a-derived-table/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  component              = "create-a-derived-table"
+  environment            = "production"
+  is-production          = "true"
+  owner                  = "data-platform:data-platform-tech@digital.justice.gov.uk"
+  infrastructure-support = "data-platform:data-platform-tech@digital.justice.gov.uk"
+  source-code            = "github.com/ministryofjustice/data-platform/terraform/aws/analytical-platform-data-production/create-a-derived-table"
```
Comment (jacobwoffenden) [inline terraform/aws/analytical-platform-data-production/create-a-derived-table/terraform.tfvars]:
```suggestion
  owner                  = "analytical-platform:analytical-platformdigital.justice.gov.uk"
  infrastructure-support = "analytical-platform:analytical-platform@digital.justice.gov.uk"
  source-code            = "github.com/ministryofjustice/analytical-platform/terraform/aws/analytical-platform-data-production/create-a-derived-table"
```

**Review by jacobwoffenden — COMMENTED** [review]
Couple of naming things, also I can see changes to components outside of this, like the production cluster, is that the result of a weird merge commit? can this be rebased?

**Review by jhpyke — APPROVED** [review]
Looking good to me

**Pull request merged**